src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java

```diff
@@ -87,6 +85,9 @@ void addGermplasmEntriesToList(Integer germplasmListId, SearchCompositeDto<Germp
 	void addGermplasmListEntriesToAnotherList(Integer destinationListId, Integer sourceListId, String programUUID,
 		SearchCompositeDto<GermplasmListDataSearchRequest, Integer> searchComposite);
 
+	GermplasmListGeneratorDTO cloneGermplasmList(Integer listId, GermplasmListGeneratorDTO listGeneratorDTO,
```

Comment:
See if you can optimize by using GermplasmListDto - with the latest changes - instead of GermplasmListGeneratorDTO which can be confusing since data and entries are not being read/saved from the DTO:
`GermplasmListDto cloneGermplasmList(Integer sourceListId, GermplasmListDto germplasmListDto, Integer loggedInUser);`